### PR TITLE
Adds README.md to conformance tests after serving->networking move

### PR DESCRIPTION
**Pull request by rafaeltello**
* Updates serving -> networking where it makes sense.
* Removes the old way of running tests, as it doesn't seem to work anymore
* Adds section on how to run locally modified conformance tests.

Please let me know if I missed anything!

### CODE REVIEW

File: test/conformance/ingress/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+   with the Ingress implementation of choice installed.
+   ```bash
+   # Set the Ingress class annotation to use in tests.
+   # Some examples:
+   #   export INGRESS_CLASS=gloo.ingress.networking.knative.dev      # Gloo Ingress
+   #   export INGRESS_CLASS=istio.ingress.networking.knative.dev     # Istio Ingress
+   #   export INGRESS_CLASS=kourier.ingress.networking.knative.dev   # Kourier Ingress
+   export INGRESS_CLASS=<your-ingress-class-annotation>
+   ```
+1. Knative Serving source code check out at `${SERVING_ROOT}`. Often this is
```
Comment (tcnghia) [inline test/conformance/ingress/README.md:27]: The test images are also moved here. We probably need a new `upload-test-images.sh`

Comment (rafaeltello) [reply test/conformance/ingress/README.md:27]: Ah! Sounds good! I didn't realize that, let me make the changes and come back.

File: test/conformance/ingress/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# Ingress Conformance Testing
+
+This directory contains Ingress conformance tests for Knative Ingress resource.
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:3]: Remove trailing whitespace:
```suggestion
This directory contains Ingress conformance tests for Knative Ingress resource.
```

File: test/conformance/ingress/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+   #   export INGRESS_CLASS=istio.ingress.networking.knative.dev     # Istio Ingress
+   #   export INGRESS_CLASS=kourier.ingress.networking.knative.dev   # Kourier Ingress
+   export INGRESS_CLASS=<your-ingress-class-annotation>
+   ```
+1. Knative Networking source code check out at `${NETWORKING_ROOT}`. Often this is
+   `$GO_PATH/src/go/knative.dev/networking`. This contains both the test images and the tests.
+   ```bash
+   export NETWORKING_ROOT=<where-you-checked-out-knative/networking>
+   ```
+1. (Recommended) Knative net-istio source code checked out. This contains an invocation
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:32]: Remove trailing whitespace:
```suggestion
1. (Recommended) Knative net-istio source code checked out. This contains an invocation
```

File: test/conformance/ingress/README.md (hunk 2 above)
Comment (mattmoor) [inline test/conformance/ingress/README.md:3]: Format markdown:
```suggestion
This directory contains Ingress conformance tests for Knative Ingress resource.
```

File: test/conformance/ingress/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+   ```bash
+   # Set the Ingress class annotation to use in tests.
+   # Some examples:
+   #   export INGRESS_CLASS=gloo.ingress.networking.knative.dev      # Gloo Ingress
+   #   export INGRESS_CLASS=istio.ingress.networking.knative.dev     # Istio Ingress
+   #   export INGRESS_CLASS=kourier.ingress.networking.knative.dev   # Kourier Ingress
+   export INGRESS_CLASS=<your-ingress-class-annotation>
+   ```
+1. Knative Networking source code check out at `${NETWORKING_ROOT}`. Often this is
+   `$GO_PATH/src/go/knative.dev/networking`. This contains both the test images and the tests.
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:28]: Format markdown:
```suggestion
1. Knative Networking source code check out at `${NETWORKING_ROOT}`. Often this
   is `$GO_PATH/src/go/knative.dev/networking`. This contains both the test
   images and the tests.
```

File: test/conformance/ingress/README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+   ```bash
+   export NETWORKING_ROOT=<where-you-checked-out-knative/networking>
+   ```
+1. (Recommended) Knative net-istio source code checked out. This contains an invocation 
+of `RunConformance` that easily allows to run tests.
+1. (For setup only) Knative Serving source code check out at `${SERVING_ROOT}`. Often this is
+   `$GO_PATH/src/go/knative.dev/serving`. This contains the `knative-testing` resources.
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:35]: Format markdown:
```suggestion
1. (Recommended) Knative net-istio source code checked out. This contains an
   invocation of `RunConformance` that easily allows to run tests.
1. (For setup only) Knative Serving source code check out at `${SERVING_ROOT}`.
   Often this is `$GO_PATH/src/go/knative.dev/serving`. This contains the
   `knative-testing` resources.
```

File: test/conformance/ingress/README.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+import (
+	"testing"
+	"knative.dev/serving/test/conformance/ingress"
+)
+
+func TestYourIngressConformance(t *testing.T) {
+	ingress.RunConformance(t)
+}
+
+```
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:101]: Format markdown:
```suggestion
```

```

File: test/conformance/ingress/README.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+### Running the tests from `net-istio` repository
+
+`net-istio` already invokes the `RunConformance` function in [`ingress_test.go`](https://github.com/knative-sandbox/net-istio/blob/master/test/conformance/ingress_test.go), so it offers a convenient place to run the tests.
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:104]: Format markdown:
```suggestion
`net-istio` already invokes the `RunConformance` function in
[`ingress_test.go`](https://github.com/knative-sandbox/net-istio/blob/master/test/conformance/ingress_test.go),
so it offers a convenient place to run the tests.
```

File: test/conformance/ingress/README.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+If `INGRESS_CLASS` is already set, then you can simply `go test ingress_test.go`
+
+## How to run tests from your local repository
+1. Clone the net-istio repository (or use any repository that invokes [`RunConformance`](./run.go)).
+1. In net-istio, add an entry to go.mod that points to your local networking folder:
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:110]: Format markdown:
```suggestion

1. Clone the net-istio repository (or use any repository that invokes
   [`RunConformance`](./run.go)).
1. In net-istio, add an entry to go.mod that points to your local networking
   folder:

```
knative.dev/networking => {YOUR NETWORKING_ROOT}
```
```

File: test/conformance/ingress/README.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+ ```
+ knative.dev/networking => {YOUR NETWORKING_ROOT}
+ ```
```
Comment (mattmoor) [inline test/conformance/ingress/README.md:114]: Format markdown:
```suggestion

```

File: test/conformance/ingress/README.md (hunk 1 above)
Comment (rafaeltello) [reply test/conformance/ingress/README.md]: Updated!

**Comment by rafaeltello** [comment]
> Produced via:
> `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)`

Thanks Matt! I was wondering how did you review this :) I'll run it next time directly.

**Comment by tcnghia** [comment]
/lgtm
/approve

Thanks a lot @rafaeltello

**Pull request merged**